cognite_toolkit/cdf_tk/load.py

```diff
@@ -419,6 +423,34 @@ def retrieve(self, ids: Sequence[T_ID]) -> T_ResourceList:
         found = [g for g in remote if g.name in ids]
         return found
 
+    def delete(self, ids: Sequence[T_Resource]) -> int:
+        # Let's prevent that we delete groups we belong to
+        try:
+            groups = self.client.iam.groups.list().data
+        except Exception as e:
+            print(
+                f"[bold red]ERROR:[/] Failed to retrieve the current service principal's groups. Aborting group deletion.\n{e}"
+            )
+            return
+        my_source_ids = []
+        for g in groups:
+            if g.source_id not in my_source_ids:
+                my_source_ids.append(g.source_id)
```

Comment:
O(1) instead of O(n) lookup
```suggestion
        my_source_ids = set()
        for g in groups:
            if g.source_id not in my_source_ids:
                my_source_ids.add(g.source_id)
```